catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy

```diff
@@ -0,0 +1,283 @@
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+class ExportCommandSpec extends spock.lang.Specification {
+
+    ExportCommand exportCommand
+
+    CatalogFramework catalogFramework
+
+    MetacardTransformer xmlTransformer
+
+    File tmpHomeDir
+
+    void setup() {
+        tmpHomeDir = File.createTempDir()
+        System.setProperty("ddf.home", tmpHomeDir.canonicalPath)
+
+        ServiceReference xmlTransformerReference = Mock(ServiceReference)
+
+        xmlTransformer = Mock(MetacardTransformer)
+
+        xmlTransformer.transform(_ as Metacard, _ as Map) >> { metacard, map ->
+            metacardToXmlTransformer(metacard, map)
+        }
+
+        BundleContext bundleContext = Mock(BundleContext) {
+            getServiceReferences(MetacardTransformer, '(id=xml)') >> [xmlTransformerReference]
+            getService(xmlTransformerReference) >> xmlTransformer
+        }
+
+        catalogFramework = Mock(CatalogFramework)
+        catalogFramework.query(_ as QueryRequest) >> { QueryRequest req ->
+            new QueryResponseImpl(req, [], 0)
+        }
+        catalogFramework.getLocalResource(_ as ResourceRequest) >> {
+            throw new ResourceNotFoundException('Could not find exception')
+        }
+
+        exportCommand = new ExportCommand(filterBuilder: new GeotoolsFilterBuilder(),
+                bundleContext: bundleContext, catalogFramework: catalogFramework)
+
+    }
+
+    void cleanup() {
+        assert tmpHomeDir?.deleteDir()
+    }
+
+    def "Test export no items"() {
+        setup:
+        exportCommand.with {
+            delete = false
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+    }
+
+    def "Test filename that already exists"() {
+        setup:
+        def file = Paths.get(System.getProperty('ddf.home'), 'filealreadyexists').toFile()
+        file.createNewFile()
+        exportCommand.with {
+            delete = false
+            output = file.canonicalPath
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
+    }
+
+    def "Test bad file name"() {
+        setup:
+        def file = Paths.get(System.getProperty('ddf.home'), 'badFilename.notazip')
+        exportCommand.with {
+            delete = false
+            output = file
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
+    }
+
+    def "Test abort command"() {
+        setup:
+        InputStream keyboardInput = new ByteArrayInputStream("n\r".getBytes('utf-8'))
+        Session session = Mock(Session) {
+            getKeyboard() >> keyboardInput
+        }
+        exportCommand.with {
+            it.delete = true
+            it.session = session
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
```

Comment:
❓ Do we really need to assert that no exception was thrown when we exception success? I would imagine that if one is thrown and we don't have the assertion, the test would still fail?